Assets/Expansion2/Replacements/citypanelpower_CQUI.lua

```diff
@@ -38,10 +40,13 @@ function OnRefresh()
 
     BASE_CQUI_OnRefresh();
 
-  -- Hide the advisor if option is disabled
-  if not Controls.PowerAdvisor:IsHidden() then
-      Controls.PowerAdvisor:SetHide( CQUI_ShowCityDetailAdvisor == false );
-  end
+    -- Hide the advisor if option is disabled
+    if not Controls.PowerAdvisor:IsHidden() then
+        Controls.PowerAdvisor:SetHide( CQUI_ShowCityDetailAdvisor == false );
+    end
+  
+    -- Hide the lens key if the power lens isn't shown
+    Controls.KeyPanel:SetHide( CQUI_AutoapplyPowerLensInCity == false );
```

Comment:
This is just for the future. I personally hate comparing booleans to true/false. What is the point of having a boolean if you don't treat is like a boolean? "someVar == false" means "not someVar" and then if you name them properly the code is much more readable.
Also, UI controls have 2 functions, SetHide and SetShow. Firaxis for some unknown reason loves using SetHide even if the action is to show the control... 
So, the line you added just boils down to:
Controls.KeyPanel:SetShow( CQUI_AutoapplyPowerLensInCity );

Reply:
Thank you for your response! I also don't really like comparing booleans to true/false, but I've been doing it here because I saw it done that way in a lot of other parts of the code, so I was trying to be consistent. Thanks for pointing out that SetShow exists though. I didn't realize that existed since, like you said, Firaxis loves using SetHide instead. I'll keep both of these in mind for the future!